
    <!DOCTYPE html>
    <html lang="en">
    <head>
    <meta charset="utf-8"/>
    <title> RNAseq report </title>
    <style>
    h1 {
    font-size: 2.5em;
    color: #385e72;
    text-align:center;
    }

    h2 {
    font-size: 1.4em;
    color: #6aabd2;
    background-color: white;
    margin-right: 50px;
    margin-left: 50px;
    padding: 5px 30px;
    }

    h3 {
    font-size: 1.2em;
    color: #b7cfdc;
    background-color: #385e72;
    margin-right: 50px;
    margin-left: 50px;
    padding: 5px 30px;
    }

    p {
    font-size: 1.2em;
    margin-right: 50px;
    margin-left: 50px;
    padding: 5px 30px;
    }
    </style>
    </head>
    <body style="background-color:#d9e4ec;font-family:verdana;">
    <p><br></p>
    <h1 >RNA-seq analysis report</h1>
    <p style="text-align:center;">This report was generated on 31/10/2022 by RASflow_EDC on iPOP-UP cluster.<br><br></p>
    <h2 id="project-name-projectname">  Project Test1</h2>
    <p><br></p>
    <h2 id="project-settings">  Analysis settings</h2>
    <p>The settings used for the analysis can be found in the <a href="logs/20221031T1606_configuration.txt">logs/20221031T1606_configuration.txt</a>.<br></p>
    
    <h2 id="SRA">FASTQ files from personal data</h2>
    <p> The FASTQ files were not downloaded from SRA, but come from your personal data. <br></p>
    
    <h2 id="fastq-quality-control">FASTQ quality control</h2>
    <p> QC report could not be found. Have you run QC before the rest of the analysis? If not, all the results might be impaired by low quality FASTQ reads. Running QC on raw sequences is highly recommended!<br></p>
    
    <h2 id="trimming">Trimming</h2>
    <p> Raw sequences were trimmed. A summary of the quality of the trimmed sequences can be found in 
    <a href="fastqc_after_trimming/report_quality_control_after_trimming.html">report_quality_control_after_trimming.html</a>. <br></p>
    
    <h2 id="mapping-and-counting">Mapping and counting</h2>
    <p> The mapping was done using HISAT2. </p>
    <p>Mapping quality is assessed in <a href="mapping_HISAT2/report_align_count_featureCounts.html">mapping_HISAT2/report_align_count_featureCounts.html</a>. </p>
    <p>This report also contains an overview of feature counting that was perfomed by featureCounts. </p>
    <p>Raw count tables are available in <a href="mapping_HISAT2/counting_featureCounts/countTables/">mapping_HISAT2/counting_featureCounts/countTables/</a> [Raw counts are not available in the online report].<br></p>
    <h2 id="exploratory-analysis-of-all-the-samples">Exploratory analysis of all the samples</h2>
    <p>To assess the quality of the experiment and the reproducibility of the replicates, please use the interactive 
    <a href="mapping_HISAT2/counting_featureCounts/Glimma/MDSPlot.html">MDS plot</a>. 
    <p> <embed type="text/html" src="mapping_HISAT2/counting_featureCounts/Glimma/MDSPlot.html" width="1000" height="900"> </p>
    <p>A static <a href="mapping_HISAT2/counting_featureCounts/PCA.pdf">principal component analysis</a> 
    for all the samples is also available,  
    as well as a <a href="mapping_HISAT2/counting_featureCounts/Heatmap_samples.pdf">heatmap with sample clustering</a>.  </p>
    <p><embed src= "mapping_HISAT2/counting_featureCounts/Heatmap_samples.pdf" type="application/pdf" width= "700" height= "700"><br></p>
    
    <h2 id="repeat-analysis">Repeat analysis</h2>
    <p> You have enable the analysis of repeated elements using featureCounts. </p>
    <p> Raw count tables for repeats are available in <a href="mapping_HISAT2/repeats_featureCounts/countTables/">mapping_HISAT2/repeats_featureCounts/countTables/</a> [Raw counts are not available in the online report].<br></p>
    <h2 id="exploratory-analysis-of-all-the-samples-for-the-repeats">Exploratory analysis of all the samples for the repeats</h2>
    <p>An interactive MDS plot based on repeat information only is shown at
    <a href="mapping_HISAT2/repeats_featureCounts/Glimma/MDSPlot.html">MDS plot</a>. 
    <p> <embed type="text/html" src="mapping_HISAT2/repeats_featureCounts/Glimma/MDSPlot.html" width="1000" height="900"> </p>
    <p>A static <a href="mapping_HISAT2/repeats_featureCounts/PCA_TE.pdf">principal component analysis</a> 
    for all the samples is also available,  
    as well as a <a href="mapping_HISAT2/repeats_featureCounts/Heatmap_samples_TE.pdf">heatmap with sample clustering</a>.  </p>
    <p><embed src= "mapping_HISAT2/repeats_featureCounts/Heatmap_samples_TE.pdf" type="application/pdf" width= "700" height= "700"><br></p>
    
            <h2 id="pairwise-differential-expression-analyses">Pairwise differential expression analyses</h2>
            <p>You have made one pairwise comparison using DESeq2. Please see below the results for this comparison.</p>
            
            <p><br></p>
            <h3 id="Comparison-between-J0_WT-and-J10_WT">Comparison between J0_WT and J10_WT for genes</h3>
            <p>The <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/regionReport/J0_WT_J10_WT/exploration.html">regionReport exploratory report</a> 
            is a good start to get an idea of the results.</p> 
            <p>Glimma interactive <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/Glimma/MDSPlot_J0_WT_J10_WT.html">MDS plot</a> 
            can help you to identify problematic samples.</p>
            <p> <embed type="text/html" src="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/Glimma/MDSPlot_J0_WT_J10_WT.html" width="1000" height="900"> </p>
            <p>To be easily shared and reused the <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/PCA_J0_WT_J10_WT.pdf">PCA</a> 
            and the 
            <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/SampleDistances_J0_WT_J10_WT.pdf">heatmap with sample clustering </a>
            are also available as PDF files.  </p> 
            <p>
            <embed src= "mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/PCA_J0_WT_J10_WT.pdf" type="application/pdf" width= "700" height= "700">
            <embed src= "mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/SampleDistances_J0_WT_J10_WT.pdf" width= "700" height= "700">
            </p>
            <p>DEA result tables (all genes or significantly differentially expressed genes) are stored in 
            <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Tables">mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Tables</a> [not available in the online report].</p>
            <p>Importantly, the results can be deeply explored thanks to <b>interactive</b>
            <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/Glimma/MDPlot_J0_WT_J10_WT.html">MD</a>
            and 
            <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/Glimma/Volcano_J0_WT_J10_WT.html">Volcano</a> 
            plots.</p>
            <p> <embed type="text/html" src="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/Glimma/Volcano_J0_WT_J10_WT.html" width="1050" height="1000"> </p>
            <p>A static <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/volcano_plot_J0_WT_J10_WT.pdf">Volcano plot</a> 
            is also available, as well as a 
            <a href="mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/heatmapTop_J0_WT_J10_WT.pdf">heatmap of the 30 top-regulated genes</a>. </p>
            <p>
            <embed src= "mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/volcano_plot_J0_WT_J10_WT.pdf" type="application/pdf" width= "700" height= "700">
            <embed src= "mapping_HISAT2/counting_featureCounts/DEA_DESeq2/Report/plots/heatmapTop_J0_WT_J10_WT.pdf" width= "700" height= "700">
            </p> 
                
            <p><br></p>
            <h3 id="Comparison-between-J0_WT-and-J10_WT">Comparison between J0_WT and J10_WT for repeats</h3>
            <p>The <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/regionReport/J0_WT_J10_WT/exploration.html">regionReport exploratory report</a> 
            is a good start to get an idea of the results.</p> 
            <p>Glimma interactive <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/Glimma/MDSPlot_J0_WT_J10_WT.html">MDS plot</a> 
            can help you to identify problematic samples.</p>
            <p> <embed type="text/html" src="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/Glimma/MDSPlot_J0_WT_J10_WT.html" width="1000" height="900"> </p>
            <p>To be easily shared and reused the <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/PCA_J0_WT_J10_WT.pdf">PCA</a> 
            and the 
            <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/SampleDistances_J0_WT_J10_WT.pdf">heatmap with sample clustering </a>
            are also available as PDF files.  </p> 
            <p>
            <embed src= "mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/PCA_J0_WT_J10_WT.pdf" type="application/pdf" width= "700" height= "700">
            <embed src= "mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/SampleDistances_J0_WT_J10_WT.pdf" width= "700" height= "700">
            </p>
            <p>DEA result tables (all repeats or significantly differentially expressed repeats) are stored in 
            <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Tables">mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Tables</a> [not available in the online report].</p>
            <p>Importantly, the results can be deeply explored thanks to <b>interactive</b>
            <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/Glimma/MDPlot_J0_WT_J10_WT.html">MD</a>
            and 
            <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/Glimma/Volcano_J0_WT_J10_WT.html">Volcano</a> 
            plots.</p>
            <p> <embed type="text/html" src="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/Glimma/Volcano_J0_WT_J10_WT.html" width="1050" height="1000"> </p>
            <p>A static <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/volcano_plot_J0_WT_J10_WT.pdf">Volcano plot</a> 
            is also available, as well as a 
            <a href="mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/heatmapTop_J0_WT_J10_WT.pdf">heatmap of the 30 top-regulated repeats</a>. </p>
            <p>
            <embed src= "mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/volcano_plot_J0_WT_J10_WT.pdf" type="application/pdf" width= "700" height= "700">
            <embed src= "mapping_HISAT2/repeats_featureCounts/DEA_DESeq2/Report/plots/heatmapTop_J0_WT_J10_WT.pdf" width= "700" height= "700">
            </p> 
                
    </body> 
    </html>
    